mlrun/track/trackers/mlflow_tracker.py

```diff
@@ -0,0 +1,449 @@
+        experiment_name = mlflow.environment_variables.MLFLOW_EXPERIMENT_NAME.get()
+
+        # Check if the user configured for matching the experiment name with the context name:
+        if getattr(
+            mlconf.external_platform_tracking, mlflow.__name__
+        ).match_experiment_to_runtime:
+            if experiment_name is not None:
+                context.logger.warn(
+                    f"`mlconf.external_platform_tracking.mlflow.match_experiment_to_runtime` is set to True but the "
+                    f"MLFlow experiment name environment variable ('MLFLOW_EXPERIMENT_NAME') is set for using the "
+                    f"name: '{experiment_name}'. This name will be overriden "
+                    f"with MLRun's runtime name as set in the MLRun configuration: '{context.name}'."
+                )
+            experiment_name = context.name
+            mlflow.set_experiment(experiment_name=experiment_name)
+
+    @classmethod
+    def post_run(cls, context: MLClientCtx):
+        """
+        Performs post-run tasks for logging MLFlow artifacts generated during the run.
+
+        :param context: Current mlrun context
+        """
+        # Get the last run (None means no run was created or found):
+        run = mlflow.last_active_run()
+        if run is None:
+            return
+
+        # Log the run:
+        MLFlowTracker._log_run(context=context, run=run)
+
+    @classmethod
+    def import_run(
+        cls,
+        project: MlrunProject,
+        pointer: str,
+        function_name: str = None,
+        handler: str = None,
+    ) -> RunObject:
+        """
+        Import a previous MLFlow experiment run to MLRun.
+
+        :param project:       The MLRun project to import the run to.
+        :param pointer:       The MLFlow `run_id` to import.
+        :param function_name: The MLRun function to assign this run to.
+        :param handler:       The handler for MLRun's RunObject
+
+        :return: The newly imported RunObject.
+        """
+        # Validate function name was given:
+        if function_name is None:
+            raise ValueError(
+                "For importing a MLFlow experiment run, a MLRun function name must be provided."
+            )
+
+        # Get the MLFlow run object:
+        run = mlflow.search_runs(
+            search_all_experiments=True,
+            filter_string=f"attributes.run_id = '{pointer}'",
+            output_format="list",
+        )
+        if not run:
+            raise ValueError(
+                "The provided run id was not found in the set MLFlow registry. Try to set the registry using "
+                "MLFlow's environment variables."
+            )
+        run = run[0]  # We are using a run id, so only one will be returned in the list.
+        # Create a run spec and metadata for creating the run object to hold the mlflow run:
+        run_spec = RunSpec(function=function_name, handler=handler)
```

Comment:
IMO it will be nicer to do it the mlrun way:
1. Extract the run object creation to a private method
2. use mlrun.new_task and RunObject.from_dict
3. Take inspiration from `mlrun.launcher.base.BaseLauncher._create_run_object`

Reply:
done